src/components/FeatureViews/Courses/CourseManagement.js

```diff
@@ -0,0 +1,490 @@
+          labelId="course-management-sort-tab"
+          id="course-management-sort-tab"
+          displayEmpty
+          value={filter}
+          onChange={handleChange}
+          classes={{ select: classes.menuSelect }}
+          input={<BootstrapInput />}
+          MenuProps={{
+            classes: { list: classes.dropdown },
+            anchorOrigin: {
+              vertical: "bottom",
+              horizontal: "left",
+            },
+            transformOrigin: {
+              vertical: "top",
+              horizontal: "left",
+            },
+            getContentAnchorEl: null,
+          }}
+        >
+          <MenuItem
+            ListItemClasses={{ selected: classes.menuSelected }}
+            value=""
+          >
+            {initialValue}
+          </MenuItem>
+          {chosenFilter}
+        </Select>
+      </FormControl>
+    </Grid>
+  );
+};
+
+const CourseManagement = () => {
+  const classes = useStyles();
+  const [sortByDate, setSortByDate] = useState("");
+  const [filterByGrades, setFilterByGrades] = useState("");
+  const [filterBySubjects, setFilterBySubjects] = useState("");
+  const [filterByInstructors, setFilterByInstructors] = useState("");
+
+  const handleChange = (event) => setSortByDate(event.target.value);
+
+  const GET_COURSES = gql`
+    query getCourses {
+      courses {
+        dayOfWeek
+        endDate
+        endTime
+        title
+        startTime
+        academicLevel
+        startDate
+        instructor {
+          user {
+            firstName
+            lastName
+            id
+          }
+        }
+        courseCategory {
+          id
+          name
+        }
+        courseId
+        id
+      }
+    }
+  `;
+
+  const { data, loading, error } = useQuery(GET_COURSES);
+
+  if (loading) return <Loading />;
+  if (error) return console.error(error.message);
+
+  const createFilteredListFromCourses = (filterCriteria) =>
+    data.courses.reduce(
+      (accumulator, currentValue) =>
+        !accumulator.some((dataExist) =>
+          filterCriteria === "course"
+            ? currentValue.courseCategory.id === dataExist.courseCategory.id
+            : currentValue.instructor.user.id === dataExist.instructor.user.id
+        )
+          ? [...accumulator, currentValue]
+          : accumulator,
+      []
+    );
+
+  const subjectList = createFilteredListFromCourses("course");
+  const instructorsList = createFilteredListFromCourses("instructor");
+
+  const checkFilter = (value, filter) => "" === filter || value === filter;
+  const sortDescOrder = (firstEl, secondEl) => (firstEl < secondEl ? -1 : 0);
+
+  const normalCourseDisplay = data.courses
+    .filter(
+      (course) =>
+        checkFilter(course.academicLevel, filterByGrades) &&
+        checkFilter(course.courseCategory.name, filterBySubjects) &&
+        checkFilter(course.instructor.user.lastName, filterByInstructors)
+    )
+    .sort((firstEl, secondEl) => {
+      switch (sortByDate) {
```

Comment:
don't use switch statements esp if you don't have a default case - they bloat up files because they take so many lines. you can use an object to do the same thing:
```
(firstEl, secondEl) => ({
   "start_date": ..something for start date,
   "end_date": ...something for end date,
}[sortByDate])
```